frontend/app/components/root/root.tsx

```diff
@@ -85,6 +92,18 @@ const messages = defineMessages({
   },
 });
 
+const getVisibleParentComment = (hash: string, childToParentComments: Record<string, string>) => {
```

Comment:
It looks like isn't the best way to find the parent of the hidden comment.
* because it trying to find parent every time even we don't have any collapsed comments (we can get a list of collapsed comments from redux)
* we can find a comment to which we trying to scroll without query to DOM, just from data representation of children comments

Reply:
We call this function only if we can't find comment in DOM. 
I don't understand how find uncollapsed parent for comment in a list of collapsed comments from redux, because we can collapse parent and save hash to grandchild.